sklearn/mixture/_gaussian_mixture_ic.py

```diff
@@ -0,0 +1,758 @@
+            corresponds to a single data point.
+
+        Returns
+        -------
+        labels : array, shape (n_samples,)
+            Component labels.
+        """
+        check_is_fitted(self, ["best_model_"], all_or_any=all)
+        labels = self.best_model_.predict(X)
+
+        return labels
+
+    def fit_predict(self, X, y=None):
+        """
+        Fit the models and predict clusters based on the best model.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            List of n_features-dimensional data points. Each row
+            corresponds to a single data point.
+
+        y : Ignored
+            Not used, present for API consistency by convention.
+
+        Returns
+        -------
+        labels : array, shape (n_samples,)
+            Component labels.
+        """
+        self.fit(X, y)
+
+        labels = self.predict(X)
+        return labels
+
+
+def _increase_reg(reg):
+    """
+    Increase regularization factor by factor of 10.
+
+    Parameters
+    ----------
+    reg: float
+        Current regularization factor.
+
+    Returns
+    -------
+    reg : float
+        Increased regularization
+    """
+    if reg == 0:
+        reg = 1e-06
+    else:
+        reg = reg * 10
+    return reg
+
+
+def _onehot_to_initial_params(X, onehot, cov_type):
+    """
+    Computes cluster weights, cluster means and cluster precisions from
+    a given clustering.
+
+    Parameters
+    ----------
+    X : array-like, shape (n_samples, n_features)
+        List of n_features-dimensional data points. Each row
+        corresponds to a single data point.
+    onehot : ndarray, shape (n_samples, n_clusters)
+        Each row has a 1 indicating cluster membership, other entries are 0.
+    cov_type : {'full', 'tied', 'diag', 'spherical'}
+        Covariance type for Gaussian mixture model
+    """
+    n = X.shape[0]
+    weights, means, covariances = _estimate_gaussian_parameters(
+        X, onehot, 1e-06, cov_type
+    )
+    weights /= n
+
+    precisions_cholesky_ = _compute_precision_cholesky(covariances, cov_type)
+
+    if cov_type == "tied":
+        c = precisions_cholesky_
+        precisions = np.dot(c, c.T)
+    elif cov_type == "diag":
+        precisions = precisions_cholesky_
+    else:
+        precisions = [np.dot(c, c.T) for c in precisions_cholesky_]
+
+    return weights, means, precisions
+
+
+def _labels_to_onehot(labels):
+    """
+    Converts labels to one-hot format.
+
+    Parameters
+    ----------
+    labels : ndarray, shape (n_samples,)
+        Cluster labels
+
+    Returns
+    -------
+    onehot : ndarray, shape (n_samples, n_clusters)
+        Each row has a single one indicating cluster membership.
+        All other entries are zero.
+    """
+    n = len(labels)
+    k = max(labels) + 1
+    onehot = np.zeros([n, k])
+    onehot[np.arange(n), labels] = 1
+    return onehot
```

Comment:
Maybe relying on [`preprocessing.LabelEncoder`](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html) or [`preprocessing.OneHotEncoder`](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html) instead would be beneficial here. 🙂